server/src/main/java/org/opensearch/index/codec/fuzzy/IndexInputImmutableLongArray.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec.fuzzy;
+
+import org.apache.lucene.store.RandomAccessInput;
+import org.apache.lucene.util.RamUsageEstimator;
+import org.opensearch.OpenSearchException;
+import org.opensearch.common.util.LongArray;
+
+import java.io.IOException;
+
+/**
+ * A Long array backed by RandomAccessInput.
+ * This implementation supports read operations only.
+ */
+class IndexInputImmutableLongArray implements LongArray {
+
+    private final RandomAccessInput input;
+    private final long size;
+
+    IndexInputImmutableLongArray(long size, RandomAccessInput input) {
+        this.size = size;
+        this.input = input;
+    }
+
+    @Override
+    public void close() {}
+
+    @Override
+    public long size() {
+        return size;
+    }
+
+    @Override
+    public synchronized long get(long index) {
+        try {
+            return input.readLong(index << 3);
```

Comment:
This is multiplying by 8 essentially.
Since we use a long array internally within this data structure, and each long is 8 bytes, we multiply by 8 to get to the correct index to read in the underlying RandomAccessInput (which by default reads byte by byte).